assets/app/scripts/services/data.js

```diff
@@ -233,6 +319,30 @@ angular.module('openshiftConsole')
     }
     return deferred.promise;
   };
+  
+// name:      API name, the unique name for the object 
+// namespace: API namespace
+// url:       External URL
+// context:   API context, (e.g. { project: "..."})
+// opts:      force - always request (default is false)
+//            http - options to pass to the inner $http call
+  DataService.prototype.getTemplate = function(name, namespace, url, context, opts) {
```

Comment:
feels odd to have the first three args be mutually exclusive
I'd rather do the following:
1. have the template controller decide whether to call `DataService.get("templates", name, {namespace:foo}` or `DataService.create("remoteTemplates", ...)`
2. Update DataService.get() to honor a `namespace` value in the passed context to avoid constructing a bogus `projectPromise`